pootle/static/js/shared/components/AutoIcon.js

```diff
@@ -53,19 +51,7 @@ class AutoIcon extends React.Component {
     };
   }
 
-  constructor(props) {
-    super();
-
-    this.state = assign({
-      saturation: 100,
-      lightness: 30,
-      abbreviationLength: 2,
-      mode: 'solid',
-      size: 20,
-    }, props);
-  }
-
-  componentDidMount() {
+  componentWillMount() {
```

Comment:
I don't see this lifecycle method being necessary: the `abbreviation` and `hue` values can be calculated in the `render` function and used right away. Having the lifecycle method removed means the component can be converted into a stateless functional component, i.e. a function.